code-challenges.test.js

```diff
@@ -10,42 +11,197 @@
-// Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
-// Expected output: [ 2, 1, -1 ]
+//a describe method that lists the name of the function OR naming of the particular test.
+describe("mix", () => {
+
+  // a test/it method, nested within the describe block, that in plain words, describes that the function does.
+  it("that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.", () => {
+
+    const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
 
+    const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+
+    //an expect method, nested within the test block, calling on the hello() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+    expect(mix(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+    expect(mix(hodgepodge2)).toEqual([ 2, 1, -1 ])
+  })
+})
 
 // b) Create the function that makes the test pass.
 
+// take in a mixed array
+// use filter to take out everything that is not a number
+// use .map to take the array with new values and get the desired math 
+// return an array with the REMAINDERS of the left over numbers
+// make it pass the test
+
+
+
+const mix = (array) => {
+    return array.filter(value => 
+    typeof (value) === "number").map(value => value % 3)
+
+}
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+
+// Expected output: [ 2, 0, -1, 0 ]
+// Expected output: [ 2, 1, -1 ]
+
 
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const cubeAndSum1 = [2, 3, 4]
-// Expected output: 99
-const cubeAndSum2 = [0, 5, 10]
-// Expected output: 1125
+// a describe method that lists the name of the function OR naming of the particular test.
+describe("square", () => {
+
+  // a test/it method, nested within the describe block, that in plain words, describes that the function does.
+  it("that takes in an array of numbers and returns the sum of all the numbers cubed.", () => {
 
+    const cubeAndSum1 = [2, 3, 4]
+    const cubeAndSum2 = [0, 5, 10]
+
+    //an expect method, nested within the test block, calling on the () function, followed by the .toEqual() matcher that checks the expected output of the function return.
+    expect(square(cubeAndSum1)).toEqual(99)
+    expect(square(cubeAndSum2)).toEqual(1125)
+  })
+})
 
 // b) Create the function that makes the test pass.
+
+// take in array
+// takes .map indexes and gets numbers cubed
+// return output then sums the new array using reduce
+
+const square = (array) => {
+    let newArr = array.map(value => value **3)
+    return newArr.reduce((a,b) => a + b)
+  
+}
+
```

Comment:
Reduce is a fun method!

Reply:
yes! I need to get more comfortable with using it. I forget about it too much, for as much a i could use it.